src/Tizen.NUI/src/internal/Layouting/GridLocations.cs

```diff
@@ -16,155 +16,297 @@
  */
 
 using System;
-using System.Collections.Generic;
+using Tizen.NUI.BaseComponents;
 
 namespace Tizen.NUI
 {
-    /// <summary>
-    /// [Draft] This internal class houses the algorithm for computing the locations and size of cells.
-    /// </summary>
-    internal class GridLocations
+    public partial class GridLayout
     {
+        private Node[] hEdgeList;
+        private Node[] vEdgeList;
+
+        private int maxRowCount;
+        private int maxColumnConut;
+        private float[] hLocations;
+        private float[] vLocations;
+        private float totalColumnWeight = 0;
+        private float totalRowWeight = 0;
+        private bool hasColumnWeight = false;
+        private bool hasRowWeight = false;
+
+        private GridChild[] gridChildren;
+
         /// <summary>
-        /// A struct holding the 4 points which make up a cell.
+        /// The nested class to represent a node of DAG.
         /// </summary>
-        public struct Cell
+        private class Node
         {
-            public int Start;
-            public int End;
-            public int Top;
-            public int Bottom;
-
-            /// <summary>
-            /// Initialize a cell with the given points.
-            /// </summary>
-            /// <param name="start">The start x coordinate.</param>
-            /// <param name="end">The end x coordinate.</param>
-            /// <param name="top">The top y coordinate.</param>
-            /// <param name="bottom">The bottom y coordinate.</param>
-
-            public Cell(int start, int end, int top, int bottom)
+            /// <summary>The start vertex with the same value as <c>Column/Row</c> child property.</summary>
+            public int Start { get; }
+
+            /// <summary>The end vertex with the same value as <c>Column+ColumnSpan/Row+RowSpan</c>.</summary>
+            public int End { get; }
+
+            /// <summary>The edge with the same value as measured width/height of child.</summary>
+            public float Edge { get; }
+
+            /// <summary>The weight with the same value as <c>ColumnWeight/RowWeight</c>.</summary>
+            public float Weight { get; }
```

Comment:
Let me say there is a container which use GridLayout and its size is 99.
And it has two child which has weight 1.

In this case, size of child should be 49.5f but the layout support only "int".
I think it makes children size to 50 or 49.

What do you think about this? I think it can also happen in LinearLayout.

Reply:
Size should be rounded. an object size will be 50(49.5). another object size will be 49(grid size(99) - first object size(50)).

Reply:
I changed your sample code and test it.

if change container HeightSpecification to Wrapcontent, the container has one less pixel then real bound box.

            View root = new View();
            root.WidthSpecification = 450;
            root.HeightSpecification = LayoutParamPolicies.WrapContent;
            root.BackgroundColor = new Color(0.8f, 0.8f, 0.8f, 1.0f);
            Window.Instance.GetDefaultLayer().Add(root);

please check.

Reply:
Thank you for the report. this issue has been fixed. sample also updated.